tests/e2e/operationstorm.go

```diff
@@ -208,4 +211,114 @@ var _ = utils.SIGDescribe("[csi-block-vanilla] Volume Operations Storm", func()
+		6. Wait for Disk to be attached to the node.
+		7. Delete Namespace  and Wait for volumes to be deleted and Volume Disk to be detached from the Node.
+	*/
+
+	ginkgo.It("[csi-block-vanilla] [csi-file-vanilla] [csi-guest] Delete namespace to confirm all volumes and pods are deleted", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By(fmt.Sprintf("Running test with VOLUME_OPS_SCALE: %v", volumeOpsScale))
+		ginkgo.By("Creating Storage Class")
+
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+			scParameters = nil
+			storagePolicyName = ""
+		} else if guestCluster {
+			ginkgo.By("CNS_TEST: Running for GC setup")
+			scParameters[svStorageClassName] = storagePolicyName
+			svcClient, svNamespace := getSvcClientAndNamespace()
+			setResourceQuota(svcClient, svNamespace, rqLimit)
+		}
+		if guestCluster {
+			storageclass, err = createStorageClass(client, scParameters, nil, "", "", true, "")
+		} else {
+			scSpec := getVSphereStorageClassSpec(storagePolicyName, scParameters, nil, "", "", false)
+			storageclass, err = client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		count := 0
+		for count < volumeOpsScale {
+			ginkgo.By("Creating PVCs using the Storage Class")
+			pvclaims[count], err = fpv.CreatePVC(client, namespace, getPersistentVolumeClaimSpecWithStorageClass(namespace, diskSize, storageclass, nil, ""))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			count++
+		}
+
+		ginkgo.By("Waiting for all claims to be in bound state")
+		persistentvolumes, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		podCount := 0
+		for podCount < volumeOpsScale {
+			ginkgo.By("Creating pod to attach PVs to the node")
+			pvclaim = pvclaims[podCount]
+			podArray[podCount], err = createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			podCount++
+		}
+
+		defer func() {
+			err = client.CoreV1().Namespaces().Delete(ctx, namespace, *metav1.NewDeleteOptions(0))
+			if !apierrors.IsNotFound(err) {
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		var vmUUID string
+		ginkgo.By("Verify the volumes are attached to the node vm")
+		podCount = 0
+		for podCount < volumeOpsScale {
+			pvclaim = pvclaims[podCount]
+			pv := getPvFromClaim(client, namespace, pvclaim.Name)
+
+			var exists bool
+			volumeID := pv.Spec.CSI.VolumeHandle
+			ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", pv.Spec.CSI.VolumeHandle, podArray[podCount].Spec.NodeName))
```

Comment:
move this just before `isVolumeAttachedToVM()` and log exactly what we are looking for, this will help in triaging product failures